### PR TITLE
JSON API: Add `auto_update_translations` feature

### PR DESCRIPTION
**Pull request by lezama**
cc @enejb

### CODE REVIEW

File: json-endpoints/jetpack/class.jetpack-json-api-themes-modify-endpoint.php [hunk 1]
```diff
@@ -64,4 +83,35 @@ function update() {
...
+			 * Pre-upgrade action
+			 *
+			 * @since 3.9.3
+			 *
+			 * @param object $theme WP_Theme object
+			 * @param array $themes Array of theme objects
+			 */
+			do_action('jetpack_pre_theme_upgrade', $theme, $this->themes);
+			// Objects created inside the for loop to clean the messages for each theme
+			$skin = new Automatic_Upgrader_Skin();
```
Comment (enejb) [inline json-endpoints/jetpack/class.jetpack-json-api-themes-modify-endpoint.php]: I think we should use our own `Automatic_Upgrader_Skin` skin.

Comment (lezama) [inline json-endpoints/jetpack/class.jetpack-json-api-themes-modify-endpoint.php]: we could, maybe on another PR?

File: class.jetpack-autoupdate.php [hunk 2]
```diff
@@ -46,6 +47,29 @@ public function autoupdate_plugin( $update, $item ) {
 		}
 		return $update;
 	}
+	
+	public function autoupdate_translation( $update, $item ) {
+		$autoupdate_themes_translations = Jetpack_Options::get_option( 'autoupdate_themes_translations', array() );
+		$autoupdate_theme_list = Jetpack_Options::get_option( 'autoupdate_themes', array() );
+		
+		// $item = {
```
Comment (lezama) [inline class.jetpack-autoupdate.php]: if we are going to leave the comment let's use `/* */`

File: class.jetpack-autoupdate.php [hunk 3]
```diff
@@ -46,6 +47,29 @@ public function autoupdate_plugin( $update, $item ) {
...
+		//  "slug":"twentyfourteen",
+		//  "language":"en_CA",
+		//  "version":"1.8",
+		//  "updated":"2015-07-18 11:27:20",
+		//  "package":"https:\/\/downloads.wordpress.org\/translation\/theme\/twentyfourteen\/1.8\/en_CA.zip",
+		//  "autoupdate":true
+		//}
+		if ( (  in_array( $item->slug, $autoupdate_themes_translations )
+	         || in_array( $item->slug, $autoupdate_theme_list ) )
+             && 'theme' === $item->type ) {
```
Comment (lezama) [inline class.jetpack-autoupdate.php]: indentation seems a little bit weird here

File: tests/php/sync/test_class.jetpack-sync-options.php [hunk 4]
```diff
@@ -136,6 +136,7 @@ public function test_sync_default_options() {
 			'jetpack_available_modules'            => 'pineapple',
 			'jetpack_autoupdate_plugins'           => 'pineapple',
 			'jetpack_autoupdate_themes'            => 'pineapple',
+			'jetpack_autoupdate_themes_translations'=> 'pineapple',
```
Comment (lezama) [inline tests/php/sync/test_class.jetpack-sync-options.php]: missing space before `=>`

**Pull request merged**